### PR TITLE
refactor: run build in container

### PR DESCRIPTION
**Pull request by eddiezane**
## Description

Currently the build process for pepr builds the code outside of the container and then copies the transpiled code in.

This PR changes the build process to run the build step inside of a container. This ensures that builds are fully reproducible and not dependent on external tooling. It also streamlines building pepr locally.

## Type of change

- [x] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/contributor-guide/#submitting-a-pull-request) followed

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -43,10 +43,6 @@ jobs:
...
           npm install -g npm
 
-          npm ci
-          npm --no-git-tag-version version ${{ github.ref_name }}
-          npm run build
-
-          docker buildx build --push --platform linux/arm64/v8,linux/amd64 --build-arg VER=${{ github.ref_name }} --tag ghcr.io/defenseunicorns/pepr/controller:${{ github.ref_name }} .
+          docker buildx build --push --platform linux/arm64/v8,linux/amd64 --build-arg PEPR_BUILD_VERSION=${{ github.ref_name }} --tag ghcr.io/defenseunicorns/pepr/controller:${{ github.ref_name }} .
 
           PEPR_VERSION=${{ github.ref_name }} npm publish --provenance --access public
```
Comment (eddiezane) [inline .github/workflows/release.yml:48]: I am not sure if `PEPR_VERSION` does anything here. I can't find any other references to it in the project.

File: Dockerfile [hunk 2]
```diff
@@ -1,35 +1,47 @@
 ### BUILD ###
 
-FROM docker.io/library/node:22-alpine@sha256:94567107148ac59f1eb2ad9b7c1db03f1a1a12d28717b29eda0535aa3bd2f71e as with-git
+ARG PEPR_BUILD_VERSION=0.0.0-development
 
-WORKDIR /app
+# crane digest cgr.dev/chainguard/node-lts:latest-dev
+FROM cgr.dev/chainguard/node-lts@sha256:49163a9f8b52f3ad9e842a16ccf0d7fa673571a272762f6483e38b90ba83985f AS build
```
Comment (eddiezane) [inline Dockerfile:6]: We'll need to make sure dependabot recognizes this as a different tag

Comment (cmwylie19) [reply Dockerfile:6]: Sounds good, we will be ready to update the `dependabot.yaml` file should it  be necessary. 

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore

**Review by cmwylie19 — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (cmwylie19) [reply .github/workflows/release.yml:48]: it could probably be `PEPR_BUILD_VERSION`

**Comment by cmwylie19** [comment]
LGTM, works, nice enhancement!

**Comment by cmwylie19** [comment]
We will be ready to update the `dependabot.yaml` should we need to ignore `PEPR_BUILD_VERSION` but i don't think it will be necessary.

**Pull request merged**